src/ProtocolZoo/ProtocolZoo.jl

```diff
@@ -208,9 +208,14 @@ end
         b_ = findfreeslot(prot.net[prot.nodeB]; randomize=prot.randomize, margin=margin)
 
         if isnothing(a_) || isnothing(b_)
-            isnothing(prot.retry_lock_time) && error("We do not yet support waiting on register to make qubits available") # TODO
-            @debug "EntanglerProt between $(prot.nodeA) and $(prot.nodeB)|round $(round): Failed to find free slots. \nGot:\n1. \t $a_ \n2.\t $b_ \n retrying..."
-            @yield timeout(prot.sim, prot.retry_lock_time)
+            if isnothing(prot.retry_lock_time)
+                @debug "EntanglerProt between $(prot.nodeA) and $(prot.nodeB)|round $(round): Failed to find free slots. \nGot:\n1. \t $a_ \n2.\t $b_ \n waiting..."
+                @yield lock(prot.nodeA.stateindices.waiter)
+                @yield lock(prot.nodeB.stateindices.waiter)
```

Comment:
I am not sure, but I think this should be `@yield lock(...) | lock(...)` so that it runs whenever there is a change to either one of them. This would need more testing for correctness though.